Orange/tests/test_table.py

```diff
@@ -53,285 +54,146 @@ def test_filename(self):
-            d[0] = ["3.14", "1", "f"]
-            almost_equal_list(d[0].values(), [3.14, "1", "f"])
+            for index, value, realValue in ((0, ["3.14", "1", "f"], [3.14, "1", "f"]),
+                                            (0, [3.15, 1, "t"], [3.15, "0", "t"]),
+                                            (np.int_(0), [3.15, 2, "f"], [3.15, 2, "f"]),
+                                            (0, data.Instance(d.domain, ["3.16", "1", "f"]), [3.16, "1", "f"]),
+                                            (np.int_(0), data.Instance(d.domain, ["3.16", 2, "t"]), [3.16, 2, "t"]),
+                                            (0, data.Instance(d.domain, ["3.16", "1", "f"]), [3.16, "1", "f"])):
+                d[index] = value
+                almost_equal_list(d[0].values(), realValue)
+
             self.assertTrue(isnan(d[0, "a"]))
-            d[0] = [3.15, 1, "t"]
-            almost_equal_list(d[0].values(), [3.15, "0", "t"])
-            d[np.int_(0)] = [3.15, 2, "f"]
-            almost_equal_list(d[0].values(), [3.15, 2, "f"])
+            d[0, "e"] = "mmmapp"
+            self.assertEqual(d[0, "e"], "mmmapp")
 
             with self.assertRaises(ValueError):
                 d[0] = ["3.14", "1"]
 
             with self.assertRaises(ValueError):
                 d[np.int_(0)] = ["3.14", "1"]
 
-            ex = data.Instance(d.domain, ["3.16", "1", "f"])
-            d[0] = ex
-            almost_equal_list(d[0].values(), [3.16, "1", "f"])
-
-            ex = data.Instance(d.domain, ["3.16", 2, "t"])
-            d[np.int_(0)] = ex
-            almost_equal_list(d[0].values(), [3.16, 2, "t"])
-
-            ex = data.Instance(d.domain, ["3.16", "1", "f"])
-            ex["e"] = "mmmapp"
-            d[0] = ex
-            almost_equal_list(d[0].values(), [3.16, "1", "f"])
-            self.assertEqual(d[0, "e"], "mmmapp")
-
     def test_slice(self):
-        import warnings
-
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
+        with catch_warnings():
+            simplefilter("ignore")
             d = data.Table("test2")
-            x = d[:3]
-            self.assertEqual(len(x), 3)
-            self.assertEqual([e[0] for e in x], [0, 1.1, 2.22])
-
-            x = d[2:5]
-            self.assertEqual(len(x), 3)
-            self.assertEqual([e[0] for e in x], [2.22, 2.23, 2.24])
-
-            x = d[4:1:-1]
-            self.assertEqual(len(x), 3)
-            self.assertEqual([e[0] for e in x], [2.24, 2.23, 2.22])
-
-            x = d[-3:]
-            self.assertEqual(len(x), 3)
-            self.assertEqual([e[0] for e in x], [2.26, 3.333, Unknown])
+            for val, realValue in ((d[:3], [0, 1.1, 2.22]),
+                                   (d[2:5], [2.22, 2.23, 2.24]),
+                                   (d[4:1:-1], [2.24, 2.23, 2.22]),
+                                   (d[-3:], [2.26, 3.333, Unknown])):
+                x = val
```

Comment:
Actually nothing smart :). I will delete it and in next lines include val where is needed instead x :)